src/main/java/gift/service/product/ProductServiceImpl.java

```diff
@@ -32,55 +33,55 @@ public List<Product> getAllProducts(String role) {
     }
 
     @Override
-    public Optional<Product> getProductById(Long id, String role) {
+    public Optional<Product> getProductById(Long id, Role role) {
         Optional<Product> opt = repo.findById(id);
         Product p = opt.orElseThrow(() -> new ProductNotFoundException(id));
-        if ("USER".equals(role) && p.hidden()) {
+        if (role == Role.USER && p.hidden()) {
             throw new ProductNotFoundException(id);
         }
         return Optional.of(p);
     }
 
     @Override
-    public Product createProduct(String name, int price, String imageUrl, String role) {
+    public Product createProduct(String name, int price, String imageUrl, Role role) {
         Product newProduct = Product.of(name, price, imageUrl);
-        if ("USER".equals(role) && isForbidden(name)) {
+        if (role == Role.USER && isForbidden(name)) {
             newProduct = newProduct.withHidden(true);
         }
         return repo.save(newProduct);
     }
 
     @Override
-    public Product updateProduct(Long id, String name, int price, String imageUrl, String role) {
+    public Product updateProduct(Long id, String name, int price, String imageUrl, Role role) {
         Product existing = repo.findById(id)
                 .orElseThrow(() -> new ProductNotFoundException(id));
-        if ("USER".equals(role) && existing.hidden()) {
+        if (role == Role.USER && existing.hidden()) {
             throw new ProductNotFoundException(id);
         }
 
         Product updated = existing.withName(name)
                 .withPrice(price)
                 .withImageUrl(imageUrl);
 
-        if ("USER".equals(role) && isForbidden(name)) {
+        if (role == Role.USER && isForbidden(name)) {
             updated = updated.withHidden(true);
         }
         return repo.save(updated);
     }
 
     @Override
-    public void deleteProduct(Long id, String role) {
+    public void deleteProduct(Long id, Role role) {
         Product existing = repo.findById(id)
                 .orElseThrow(() -> new ProductNotFoundException(id));
-        if ("USER".equals(role) && existing.hidden()) {
+        if (role == Role.USER && existing.hidden()) {
             throw new ProductNotFoundException(id);
         }
         repo.deleteById(id);
     }
 
     @Override
-    public void hideProduct(Long id, String role) {
-        if ("USER".equals(role)) {
+    public void hideProduct(Long id, Role role) {
+        if (role == Role.USER) {
```

Comment:
```suggestion
        if (role.isUser()) {
```
- 어떤 객체든 데이터 홀더로 사용하기 보단, 메시지를 요청해보려고 해보세요.